gorp.go

```diff
@@ -599,50 +601,26 @@ func insert(m *DbMap, exec SqlExecutor, list ...interface{}) error {
 	return nil
 }
 
-func exec(e SqlExecutor, query string, args ...interface{}) (sql.Result, error) {
-	executor, ctx := extractExecutorAndContext(e)
-
-	if ctx != nil {
-		return executor.ExecContext(ctx, query, args...)
-	}
-
-	return executor.Exec(query, args...)
-}
-
-func prepare(e SqlExecutor, query string) (*sql.Stmt, error) {
-	executor, ctx := extractExecutorAndContext(e)
-
-	if ctx != nil {
-		return executor.PrepareContext(ctx, query)
-	}
+func exec(ctx context.Context, e SqlExecutor, query string, args ...interface{}) (sql.Result, error) {
+	executor := extractExecutor(e)
```

Comment:
Why is this step necessary? The SqlExecutor already has an ExecContext (or Exec, if I have my druthers) method on it. Same for `queryRow` and `query` below.

Reply:
Good point on this. I pulled on this thread and it allowed me to remove all uses of extractExecutor but one, and I inlined that remaining use.